sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java

```diff
@@ -663,9 +711,15 @@ private Charset determineCharsetFromHttpHeader(final Message message) {
                         // NOTE: returning null in this case is ok as determineCharsetFromMessage() will then try to
                         //       use other clues to determine the charset and hopefully succeed in correctly decoding
                         //       the Message before storing it in the DB.
-                        this.testRunObserver.invalidateTestRun(
-                                String.format("Encountered invalid/unknown charset '%s' in HTTP Header", charsetName),
-                                e);
+                        if (this.summarizeMessageEncodingErrors) {
+                            this.testRunObserver.invalidateTestRunWithoutReason();
```

Comment:
I am not quite sure if this is the right place to invalidate the test run. If the test run is invalidated without any reason, the user gets the message: "This test run has been invalidated because:" without any further information. Wouldn't it also be an option to invalidate the test run inside TestSuite::printVerdict with a reason?

Reply:
You are right, that's better. Changed.